src/platform/datapath_winkernel.c

```diff
@@ -2034,8 +2101,34 @@ QuicDataPathSocketReceive(
                     goto Drop;
                 }
 
+                if (RecvContext->ProcContext->OutstandingStoredBytes > 256000) {
+                    //
+                    // Perform a copy
+                    //
+                    RecvContext->IsCopiedBuffer = TRUE;
+                    RecvContext->BufferPoolIndex = DataLength > 4096 ? 1 : 0;
+                    RecvContext->DataBufferStart =
+                        (uint8_t*)QuicPoolAlloc(
+                            &RecvContext->ProcContext->RecvBufferPools[RecvContext->BufferPoolIndex]);
+                    if (RecvContext->DataBufferStart == NULL) {
```

Comment:
Idea: should we pend if we can't alloc to copy? Or just drop

Reply:
Since thats an OOM scenario, I'd drop. That way going OOM couldn't be an attack vector to lock up all kernel NBLs.